### PR TITLE
feat: HIP-904 maxAutoAssociations updates

### PR DESCRIPTION
**Pull request by MiroslavGatsanoga**
**Description**:

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Some unexpected files are present in the PR.
This should be limted to automaxassocaitions and TokenReject

**Comment by Neeharika-Sompalli** [comment]
> Some unexpected files are present in the PR. This should be limted to automaxassocaitions and TokenReject

yes @Nana-EC . Once it is synced back with base branch they should be resolved

**Review by Nana-EC — DISMISSED** [review]

**Comment by netopyr** [comment]
We plan to implement the changes for maxAutoAssociations only in the upcoming release. The protobuf modifications need to be limited accordingly.

**Review by Neeharika-Sompalli — DISMISSED** [review]
LGTM! Thanks @MiroslavGatsanoga

File: services/response_code.proto [hunk 1]
```diff
@@ -1437,4 +1437,16 @@ enum ResponseCodeEnum {
    * grpc_certificate_hash is invalid
    */
   INVALID_GRPC_CERTIFICATE = 345;
+
+  /**
+   * The maximum automatic associations value is not valid.<br/>
+   * The most common cause for this error is a value less than `-1`.
+   */
+  INVALID_MAX_AUTO_ASSOCIATIONS = 346;
+
```
Comment (Neeharika-Sompalli) [inline services/response_code.proto:1446]: May be this is not needed for this release ?

File: services/crypto_create.proto [hunk 2]
```diff
@@ -128,8 +128,12 @@ message CryptoCreateTransactionBody {
...
 
     /**
-     * The maximum number of tokens that an Account can be implicitly associated with. Defaults to 0
-     * and up to a maximum value of 1000.
+     * The maximum number of tokens that can be auto-associated with the account.<br/>
+     * If this is less than or equal to `used_auto_associations`, or 0, then this account
+     * MUST manually associate with a token before transacting in that token.<br/>
+     * This value MAY also be `-1` to indicate no limit.<br/>
+     * This value MUST NOT be less than `-1`.<br/>
+     * By default this value is 0 for accounts except for auto-created accounts which default -1.
```
Comment (Neeharika-Sompalli) [inline services/crypto_create.proto:136]: And these documentations, since @netopyr  mentioned `maxAutoAssociations` are pushed to next release.

**Review by netopyr — CHANGES_REQUESTED** [review]
Please reduce this PR to changes that are required for release 0.51.

**Review by jsync-swirlds — APPROVED** [review]

**Review by Nana-EC — APPROVED** [review]

**Review by netopyr — APPROVED** [review]
LGTM

**Pull request merged**